### DISCUSSION TITLE
Customize Script Choosing

### DISCUSSION THREAD

**Discussion by rwieruch** [vercel/turborepo#3917] [Help] [answered]
As far as I know Turborepo simply picks up all the `dev` scripts in each package.json when I run `turbo run dev`. Is there any way to specify a custom script choosing like "On `turbo run dev` pick up `dev:mode1` in package-one/ and `dev:mode2` in package-two/"?

**Comment by mehulkar** [comment, marked as answer]
This is a feature! `turbo run dev` is meant to run `dev` in all your workspaces that have that script.

You have some options though:

1. You can setup your `package-one/package.json` to have the `dev` script _call_ the `dev:mode1` (and similarly in `package-two`) like so:

    ```jsonc
    {
      // ...snipped...
      "scripts": {
        "dev": "npm run dev:mode1"
      }
    }
    ```

1. You can setup your `turbo.json` with some clever `dependsOn`:

    ```json
    {
      "pipeline": {
        "dev": {
          "dependsOn": ["dev:mode1", "dev:modes"]
        }
      }
    }
    ```
    If package-one only implements `dev:mode1` and `package-two` only implements `dev:mode2` (and neither implement `dev`), then you should get the behavior you're looking for. 

    (Note: I wouldn't _really_ recommend this approach as it feels a bit brittle, but it might work for you.)
